SDL error window was too little for big messages

galegosimpatico opened the pull request:
The important thing now for me for this was just to have you read this comment
in 1c3c9149ebor (reachable via this link https://github.com/anura-engine/anura/commit/1c3c9149ebb7ef76b56f3fd62515f09d2df4d452), I do not know if this still applies.

Just discard silently if you think it makes not much sense now. I was clearing up my trunk for it to be paired with upstream.

galegosimpatico committed:
Added comments for building under Debian Wheezy.

Things will become easier when Jessie is released...

galegosimpatico committed:
Detail INSTALL changes.

galegosimpatico committed:
Care more on error formatting on type errors.

variant.cpp: Providing 'generate_error_2(2)' almost same as
'generate_error(1)', but 'generate_error_2(2)' accepts two different
messages, one to be printed to console, and the other one to
be passed to 'SDL_ShowSimpleMessageBox(4)'. 'generate_error(1)' prints
the same string both to console and the SDL box, this leads to a badly
stacked SDL box if the message is very long and/or the screen is
small. 'generate_error_2(2)' can be used to get wide console output
while caring better on message stacking in an SDL box.
'SDL_ShowSimpleMessageBox(4)' handles badly large strings.

variant.cpp: 'generate_error_2(2)' does the same as
'generate_error(1)' but relying on a new 'ASSERT_LOG_3(3)' instead of
'ASSERT_LOG(2)'.

asserts.hpp, asserts.cpp: Providing 'ASSERT_LOG_3(3)' to be able to
log assertions with different strings for console and SDL.
'ASSERT_LOG_3(3)' does the same as 'ASSERT_LOG(2)' but passing down
the arguments to 'report_assert_msg_2(2)' instead of
'report_assert_msg(1)'.

asserts.hpp, asserts.cpp: Providing 'report_assert_msg_2(2)' that does
the same as 'report_assert_msg(1)' but instead of printing the same
message to console and the SDL simple box, receives two different
strings, the first to be printed to console and the second to be
passed to 'SDL_ShowSimpleMessageBox(4)'. This is because
'SDL_ShowSimpleMessageBox(4)' handles badly too wide messages, while
text logs and console output fits naturally and safely lines of
arbitrary length.

galegosimpatico committed:
Four missing 'cd ../' in INSTALL.

But assuming bash or zsh is available, change 'cd DIR/' to 'pushd DIR'
and 'cd ../' to 'popd'.

src/variant.cpp: Beautify two assertions.

galegosimpatico committed:
Merge remote-tracking branch 'upstream/trunk' into trunk

galegosimpatico committed:
Detail INSTALL document update.

galegosimpatico committed:
Detail .gitignore update - IDEs.

Miscellanous IDEs metadata directories and documents.

galegosimpatico committed:
Merge branch 'trunk' of https://github.com/anura-engine/anura into trunk

galegosimpatico committed:
Adapt unofficial clone to upstream changes.

Adapt custom function 'report_assert_msg_2(2)' (report function that
receives two strings instead of a single one, one is printed to the
console and the other one is printed to the assertion SDL simple box)
to upstream changes.

galegosimpatico committed:
Add small note for building under Jessie.

galegosimpatico committed:
Merge remote branch 'upstream/trunk' into 'trunk'.

Merge remote-tracking branch 'upstream/trunk' into 'trunk'.

galegosimpatico committed:
Adapt local fork to upstream changes.

galegosimpatico committed:
Merge remote branch 'upstream/trunk' into 'trunk'.

Merge remote-tracking branch 'upstream/trunk' into 'trunk'.

galegosimpatico committed:
Merge remote-tracking br. 'u.../t...' into 't...'.

Merge remote-tracking branch 'upstream/trunk' into 'trunk'.

galegosimpatico committed:
Merge branch 'trunk' of 'upstream' into 'trunk'.

'origin' set to 'https://galegosimpatico:********@github.com/galegosimpatico/anura'.

'upstream' set to 'https://github.com/anura-engine/anura'.

galegosimpatico committed:
Merge branch 'trunk' into 'my-trunk'.

ghost commented:
There is now a problem in Debian Wheezy build (as stated there) regarding the OpenGL ES API now in use by Anura being 3.1 instead of 2.0. I'll be attempting to fix it.

ghost commented:
Anura being able to compile and link with no error on Debian Wheezy when manually installing `GLES3/gl31.h` and `glew-1.13.0`. Debian Wheezy ships GLEW 1.7 by its default package repositories, and 1.7 is now out of support by Anura. But GLEW 1.13 is straightforward to install to Debian Wheezy while making Anura compile and link OK (with a little bit of `Makefile` tweaking).

When running, seems to be crashing with:

    CRITICAL: WindowManager.cpp:284 ASSERTION FAILED: Failed to create renderer: Couldn't
    find matching render driver

The message happens no matter the loaded module. Maybe the build is not correct, but I would say that same crash message happened to my Debian Jessie mileage...

Deselecting `libgl1-mesa-glx` and substituting it with `libgl1-mesa-swx11` (software based Mesa?) makes it fail when trying to run:

    src/main.cpp:1022
    -----------------
    int swap_result = SDL_GL_SetSwapInterval(1);

Function documentation: https://wiki.libsdl.org/SDL_GL_SetSwapInterval. I tried all three possible arguments to the function, not really knowing what I was doing, yet any argument were causing the function to fail.

By commenting the operation, Anura gets able to run (well...) frogatto, citadel and elemedals, but not any of the included demo modules nor the tictactoe external demo nor duxduo or metroiduh.

Module elemedals was runnable (not playable though), closable, and extremely and maybe increasingly slow, sometimes eventually seemingly becoming freezed. While this, the main execution thread was using 100% CPU.

Frogatto was runnable and closable with status success, and even slower than elemedals.

All this given, I am not 100% sure the Debian Wheezy build is already updated to upstream, but I consider it probably enough correct, and the problems I had, as a hardware related issue.

In case you wanted further updates to the Debian Wheezy install guide.

I plan on improve the Debian Jessie install guide, possibly with another separate pull request.

Cheers and regards,

galegosimpatico committed:
Debian Wheezy install guide up to date.

galegosimpatico committed:
Fix the Debian Jessie build step by step guide.

ghost commented:
Fixed the Jessie build guide that was missing an explicit advice on using a custom installed Boost 1.56.0 downloaded from official Boost sources, instead of the Boost distributed version by Debian Jessie, that is 1.55.0, no longer compatible given since March 2016 there is the `src/kre/AlignedAllocator.hpp` dependent on a Boost header that is present in Boost 1.56.0 but not existing in Boost 1.55.0 previously.

galegosimpatico committed:
Detail fixes for the Jessie build guide.

Comment in the general section the need of Boost 56. Boost 50 is not
currently supported.

ghost commented:
Will treat each addressed issue separately.
